executor/join.go

```diff
@@ -504,10 +505,13 @@ func (e *HashJoinExec) buildHashTableForList(innerResultCh <-chan *chunk.Chunk)
 		allTypes:  allTypes,
 		keyColIdx: innerKeyColIdx,
 	}
-	initList := chunk.NewList(allTypes, e.initCap, e.maxChunkSize)
-	e.rowContainer = newHashRowContainer(e.ctx, int(e.innerEstCount), hCtx, initList)
+	e.rowContainer = newHashRowContainer(e.ctx, int(e.innerEstCount), hCtx)
 	e.rowContainer.GetMemTracker().AttachTo(e.memTracker)
 	e.rowContainer.GetMemTracker().SetLabel(innerResultLabel)
+	if config.GetGlobalConfig().OOMUseTmpStorage {
+		actionSpill := e.rowContainer.ActionSpill()
+		e.ctx.GetSessionVars().StmtCtx.MemTracker.SetActionOnExceedWithFallback(actionSpill)
```

Comment:
The function name is really confusion. How about this:
```go
stmtMemTracker := e.ctx.GetSessionVars().StmtCtx.MemTracker
oldAction := stmtMemTracker.GetActionOnExceed()
stmtMemTracker.SetActionOnExceed(actionSpill)
stmtMemTracker.SetActionOnExceedAgain(oldAction)
```

Reply:
We must keep all the SetActionOnExceedWithFallback operations in the same critical area.
Do you came up a better name?

Reply:
How about using a list of actions?
For the first time, the actions[0] will be triggered, and for the second time, the actions[1] will be triggered, and so on...
Then we don't need to introduce the concept `fallback`.

Reply:
We have some logical here to handle whether to trigger fallback action.
https://github.com/pingcap/tidb/pull/12067/files#diff-4f7e05240341383641662778116ddf13R287-R291
For example, when the action is spilling right now, it should not call its fallback action.